ast/substitute/Substitute.cc

```diff
@@ -6,9 +6,9 @@
 namespace sorbet::ast {
 
 namespace {
-class SubstWalk {
+template <typename T> class SubstWalk {
```

Comment:
I figured that this is an OK use of a template since it's confined to this file. If not, I can introduce a level of indirection via an interface.